contrib/storage-hbase/src/main/java/org/apache/drill/exec/store/hbase/HBaseRecordReader.java

```diff
@@ -124,6 +124,11 @@ public HBaseRecordReader(Connection connection, HBaseSubScan.HBaseSubScanSpec su
     } else {
       rowKeyOnly = false;
       transformed.add(ROW_KEY_PATH);
+      /* DRILL-5137 - optimize count(*) queries on MapR-DB Binary tables */
```

Comment:
This branches into the else part of
`if (!isStarQuery()) {`

Can you verify if a query can be both Star and Skip query at the same time when count(*) has been requested?